src/main/java/mate/academy/onlinebookstore/controller/CartController.java

```diff
@@ -0,0 +1,65 @@
+package mate.academy.onlinebookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.onlinebookstore.dto.item.ItemAddRequestDto;
+import mate.academy.onlinebookstore.dto.shoppingcart.CartResponseDto;
+import mate.academy.onlinebookstore.service.shoppingcart.CartService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "ShoppingCart management", description = "Endpoints for managing shopping carts.")
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+public class CartController {
+    private final CartService cartService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    @Operation(summary = "Retrieve user's shopping cart.")
+    public CartResponseDto viewCart(Pageable pageable) {
+        return cartService.getCart(pageable);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Add book to user's shopping cart.")
+    public CartResponseDto addBook(
+            @RequestBody @Valid ItemAddRequestDto requestDto, Pageable pageable) {
+
+        return cartService.addBookToCart(requestDto, pageable);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PutMapping("/items/{cartItemId}")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Update number of books to purchase.")
+    public CartResponseDto updateItem(
+            @RequestParam int quantity, @PathVariable Long cartItemId, Pageable pageable) {
```

Comment:
create dto instead of int quantity
```suggestion
            @RequestParam int quantity, @PathVariable Long cartItemId) {
```